testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/AbstractWebFailoverTestCase.java

```diff
@@ -159,8 +159,11 @@ private void testFailover(Lifecycle lifecycle, URL baseURL1, URL baseURL2, URL b
                 Assert.assertEquals(HttpServletResponse.SC_OK, response.getStatusLine().getStatusCode());
                 Assert.assertEquals(value++, Integer.parseInt(response.getFirstHeader(SimpleServlet.VALUE_HEADER).getValue()));
                 Map.Entry<String, String> entry = parseSessionRoute(response);
-                // Ensure routing is not changed on subsequent query
-                Assert.assertNull(entry);
+                if (entry != null) {
```

Comment:
What if the entry is null ?

I saw comment on WFLY-11241 - Looks like while the transfer is still in progress, the session might be pinned from original owner node1 to node2 and then to node3. Thus the checks need to be more permissive.

Shouldn't the test wait for the transfer to finish ?